tensorflow-examples/src/main/java/org/tensorflow/model/examples/cnn/fastrcnn/FasterRcnnInception.java

```diff
@@ -287,35 +287,59 @@ public static void main(String[] params) {
-                                    float detectionScore = detectionScores.getFloat(0, n);
-                                    //only include those classes with detection score greater than 0.3f
-                                    if (detectionScore > 0.3f) {
-                                        float classVal = detectionClasses.getFloat(0, n);
-                                        //TODO tf.image.drawBoundingBoxes
-                                        int x1 = (int) (shapeArray[1] * detectionBoxes.getFloat(0, n, 1));
-                                        int y1 = (int) (shapeArray[0] * detectionBoxes.getFloat(0, n, 0));
-                                        int x2 = (int) (shapeArray[1] * detectionBoxes.getFloat(0, n, 3));
-                                        int y2 = (int) (shapeArray[0] * detectionBoxes.getFloat(0, n, 2));
-                                        graphics2D.setPaint(Color.RED);
-                                        graphics2D.setStroke(new BasicStroke(5));
-                                        graphics2D.drawRect(x1, y1, x2 - x1, y2 - y1);
-                                        graphics2D.setPaint(Color.BLACK);
-                                        //add a label with percentage score
-                                        graphics2D.drawString(cocoTreeMap.get(classVal - 1) + " " +
-                                                (new DecimalFormat("#.##").format(detectionScore * 100)) +
-                                                "%", x1, y1);
-                                    }
+                            ArrayList<FloatNdArray> boxArray = new ArrayList<>();
+                            //TODO tf.image.combinedNonMaxSuppression
+                            for (int n = 0; n < numDetects; n++) {
+                                //put probability and position in outputMap
+                                float detectionScore = detectionScores.getFloat(0, n);
+                                //only include those classes with detection score greater than 0.3f
+                                if (detectionScore > 0.3f) {
+                                    boxArray.add(detectionBoxes.get(0, n));
+                                }
+                            }
+                            //2-D. A list of RGBA colors to cycle through for the boxes.
+                            Operand<TFloat32> colors = tf.constant(new float[][]{
+                                    {0.9f, 0.3f, 0.3f, 0.0f},
+                                    {0.3f, 0.3f, 0.9f, 0.0f},
+                                    {0.3f, 0.9f, 0.3f, 0.0f}
+                            });
+                            Shape boxesShape = Shape.of(1, boxArray.size(), 4);
+                            int boxCount = 0;
+                            //3-D with shape `[batch, num_bounding_boxes, 4]` containing bounding boxes
+                            try (TFloat32 boxes = TFloat32.tensorOf(boxesShape)) {
```

Comment:
I think both `boxes` and `outputImagePath` should be placeholders and then those values should be passed in. That makes it simpler to reuse the image writing portion of the graph. The way it currently is requires a new bit of graph to be added each time the image is written out, or if the boxes change. For demo code I think showing the reusable version is better because people are likely to copy this code into their own projects.

Reply:
Done